### PR TITLE
Add Diagram

### PR DESCRIPTION
**Pull request by LanaDX**

### CODE REVIEW

File: interface-elements-for-web/articles/diagram/first-look-at-the-diagram.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+- The **History** toolbar allows you to undo and redo your actions. On mobile devises the toolbar additionally displays the **Show Toolbox** button that shows/hides the **Toolbox**.
+
+    ![Show Toolbox Button](../../images/diagram-history-and-toolbox-button.png)
+
+- The **Toolbox** contains shapes available in the **Diagram**. Select the **View** > **Settings** > **Show Toolbox** command to switch the **Toolbox** visibility. On mobile devises the **Show Toolbox** button is additionally displayed in the **History** toolbar.
+
+- The **Properties** toolbar allows you to [customize shape and connector style settings](create-a-new-chart.md#customize-shape-and-connector-style-settings), change the [chart layout](chart-layout.md), and specify [page settings](page-and-view-settings.md). Click the **Properties** button to show and hide the toolbar.
+
+- The **View** toolbar contains commands related to **Diagram**'s [view settings](page-and-view-settings.md).
```
Comment (dirk-pieterse) [inline interface-elements-for-web/articles/diagram/first-look-at-the-diagram.md:19]: "to **Diagram**'s" the? a?

File: interface-elements-for-web/articles/diagram/first-look-at-the-diagram.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+- The **Context Toolbox** appears when you draw a connector from a shape and release it without it being connected to another shape. Click a shape to create it at the end of the connector. Note, this toolbox can contain a reduced set of shapes. See the **Toolbox** for the full list of available shapes.
```
Comment (dirk-pieterse) [inline interface-elements-for-web/articles/diagram/first-look-at-the-diagram.md:21]:
```suggestion
- The **Context Toolbox** appears when you draw a connector from a shape and release it without connecting it to another shape. Click a shape to create it at the end of the connector. This toolbox can contain a reduced set of shapes. See the **Toolbox** for the full list of available shapes.

```

Comment (dirk-pieterse) [reply interface-elements-for-web/articles/diagram/first-look-at-the-diagram.md:21]: Why do you add "Note, this toolbox can contain a reduced set of shapes. See the **Toolbox** for the full list of available shapes"?

File: interface-elements-for-web/articles/diagram/first-look-at-the-diagram.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+    ![Context Toolbox](../../images/diagram-context-toolbox.png)
+
+- The **Context Menu** appears when you right-click a chart element(s).
+
+    ![Context Menu](../../images/diagram-context-menu.png)
+
+    On mobile devises tap a chart element to display the **Context Menu**.
```
Comment (dirk-pieterse) [inline interface-elements-for-web/articles/diagram/first-look-at-the-diagram.md:29]:
```suggestion
    On mobile devises, tap a chart element to display the **Context Menu**. 

```

File: interface-elements-for-web/articles/diagram/first-look-at-the-diagram.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+- The **Properties** toolbar allows you to [customize shape and connector style settings](create-a-new-chart.md#customize-shape-and-connector-style-settings), change the [chart layout](chart-layout.md), and specify [page settings](page-and-view-settings.md). Click the **Properties** button to show and hide the toolbar.
+
+- The **View** toolbar contains commands related to **Diagram**'s [view settings](page-and-view-settings.md).
+
+- The **Context Toolbox** appears when you draw a connector from a shape and release it without it being connected to another shape. Click a shape to create it at the end of the connector. Note, this toolbox can contain a reduced set of shapes. See the **Toolbox** for the full list of available shapes.
+
+    ![Context Toolbox](../../images/diagram-context-toolbox.png)
+
+- The **Context Menu** appears when you right-click a chart element(s).
```
Comment (dirk-pieterse) [inline interface-elements-for-web/articles/diagram/first-look-at-the-diagram.md:25]:
```suggestion
- The **Context Menu** appears when you right-click chart elements.

```

File: interface-elements-for-web/articles/diagram/create-a-new-chart.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+1. Drag a shape from the **Toolbox** onto the page.
+
+1. Double click the shape to add/edit its text.
+
+![Add a Shape](../../images/diagram-add-shape.gif)
+
+## Connect Shapes
+
+1. Add one more shape to the chart.
```
Comment (dirk-pieterse) [inline interface-elements-for-web/articles/diagram/create-a-new-chart.md:18]:
```suggestion
1. Add another shape to the chart.

```

File: interface-elements-for-web/articles/diagram/create-a-new-chart.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+1. Hover a start shape to display its connection points. 
+
+1. Hover a point, press the left mouse button and drag the connector to another shape's connection point. Release the mouse button. Now, if you move a shape, the connector will stretch to it.
```
Comment (dirk-pieterse) [inline interface-elements-for-web/articles/diagram/create-a-new-chart.md:22]:
```suggestion
1. Hover a point, press the left mouse button, and drag the connector to another shape's connection point, and release the mouse button. Now, if you move a shape, the connector stretches to it.

```

File: interface-elements-for-web/articles/diagram.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
+---
+title: Diagram
+author: Svetlana Nikulina
+---
+# Diagram
+
+The **Diagram** allows you to design diagrams, flowcharts, and org charts online.
```
[the remark this answers is not in the document]
Comment (LanaDX) [reply interface-elements-for-web/articles/diagram.md:7]: organizational charts

File: interface-elements-for-web/articles/diagram/work-with-chart-elements.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+- **By mouse**: press the Ctrl+Shift and drag an element to clone it.
+
+- **By keyboard**: select an element and press the Ctrl+Shift+arrow key to create a copy next to the original element in the arrow direction.
+
+![Clone a Shape](../../images/diagram-clone-element.gif)
+
+
+## Link Connector to a Shape
+
+1. Hover a shape by mouse pointer to display connection points.
```
[the remark this answers is not in the document]
Comment (LanaDX) [reply interface-elements-for-web/articles/diagram/work-with-chart-elements.md:54]: Updated:
Hover the mouse cursor over a shape to display its connection points.

File: interface-elements-for-web/articles/diagram/chart-layout.md [hunk 9]
```diff
@@ -0,0 +1,19 @@
+---
+title: Chart Layout
+author: Svetlana Nikulina
+---
+# Chart Layout
+The **Diagram** can create a chart layout based on a predefined algorithm.
+
+- The **Tree** layout algorithm is used to draw hierarchical diagrams. It arranges shapes in a tree structure and aligns them evenly based on their hierarchical level.
+
+- The **Layered** layout algorithm is used to draw layered graphs. It minimizes the number of connector crossings and spreads shapes evenly.
```
[the remark this answers is not in the document]
Comment (LanaDX) [reply interface-elements-for-web/articles/diagram/chart-layout.md:10]: updated to:
It spreads shapes in horizontal rows or layers and minimizes the number of connector crossings.

File: interface-elements-for-web/articles/diagram/first-look-at-the-diagram.md [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+
+    ![Show Toolbox Button](../../images/diagram-history-and-toolbox-button.png)
+
+- The **Toolbox** allows you to add shapes to your chart. Select the **View** > **Settings** > **Show Toolbox** command to switch the **Toolbox** visibility. On mobile devises, the **Show Toolbox** button is additionally displayed in the **History** toolbar.
+
+- The **Properties** toolbar allows you to [customize shape and connector style settings](create-a-new-chart.md#customize-shape-and-connector-style-settings), change the [chart layout](chart-layout.md), and specify [page settings](page-and-view-settings.md). Click the **Properties** button to show and hide the toolbar.
+
+- The **View** toolbar contains commands related to the **Diagram**'s [view settings](page-and-view-settings.md).
+
+- The **Context Toolbox** allows you to [create a shape at the end of a connector](work-with-chart-elements.md#create-a-shape-via-the-context-toolbox) in one click.
```
Comment (dirk-pieterse) [inline interface-elements-for-web/articles/diagram/first-look-at-the-diagram.md:21]: Again, I'm not sure how "the **Context Toolbox** allows you to" do something

Comment (dirk-pieterse) [reply interface-elements-for-web/articles/diagram/first-look-at-the-diagram.md:21]: in one click?

File: interface-elements-for-web/articles/diagram/work-with-chart-elements.md [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+2. Drag a point to move a line segments.
+
+### Straight Connectors
+
+You can add line segments to connectors.
+
+1. Select a connector to display its key points.
+2. Drag a middle point to add a line segments.
+
+To remove a key point, move it on line with nearest points.
```
Comment (dirk-pieterse) [inline interface-elements-for-web/articles/diagram/work-with-chart-elements.md:119]: on line?

File: interface-elements-for-web/articles/diagram/work-with-chart-elements.md [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+
+1. Select a connector to display its key points.
+2. Drag a point to move a line segments.
+
+### Straight Connectors
+
+You can add line segments to connectors.
+
+1. Select a connector to display its key points.
+2. Drag a middle point to add a line segments.
```
Comment (dirk-pieterse) [inline interface-elements-for-web/articles/diagram/work-with-chart-elements.md:117]: a line segments? 'a' plural? Did you proofread these files?

File: interface-elements-for-web/articles/diagram/work-with-chart-elements.md [hunk 13]
```diff
@@ -0,0 +1,121 @@
...
+
+![Context Toolbox](../../images/diagram-context-toolbox-with-container.gif)
+
+## Modify a Connector
+
+The **Diagram** can draw orthogonal and straight connectors. 
+
+![Line Type](../../images/diagram-connector-line-type.png)
+
+Specify a connector's line type on the **Style** tab in the **Properties** toolbar.
```
Comment (dirk-pieterse) [inline interface-elements-for-web/articles/diagram/work-with-chart-elements.md:101]:
```suggestion
Specify a connector's line type in the **Style** tab on the **Properties** toolbar.

```

File: interface-elements-for-web/articles/diagram/work-with-chart-elements.md [hunk 14]
```diff
@@ -0,0 +1,121 @@
...
+The **Context Toolbox** appears when you drag a connector from a shape and drop it without connecting it to another shape. 
+
+Select a shape to create it at the end of the connector. 
+
+![Context Toolbox](../../images/diagram-context-toolbox.gif)
+
+The **Context Toolbox** allows you to create a shape inside a container. 
+
+1. Drag a connector from a shape while pressing Ctrl and release the connector inside a container. 
+2. Select a shape you want to add. The shape is placed in the container's nested shape collection.
```
Comment (dirk-pieterse) [inline interface-elements-for-web/articles/diagram/work-with-chart-elements.md:91]:
```suggestion
2. Select the shape you want to add. The shape is placed in the container's nested shape collection.

```

File: interface-elements-for-web/articles/diagram/work-with-chart-elements.md [hunk 15]
```diff
@@ -0,0 +1,121 @@
...
+Drag a shape close to a connector's end to link the shape to the connector.
+
+![Link a Shape to Connector](../../images/diagram-link-shape-to-connector.gif)
+
+
+## Create a Shape via the Context Toolbox
+
+The **Context Toolbox** appears when you drag a connector from a shape and drop it without connecting it to another shape. 
+
+Select a shape to create it at the end of the connector.
```
Comment (dirk-pieterse) [inline interface-elements-for-web/articles/diagram/work-with-chart-elements.md:84]:
```suggestion
Select a shape to insert it at the end of the connector. 

```

File: interface-elements-for-web/articles/diagram/work-with-chart-elements.md [hunk 16]
```diff
@@ -0,0 +1,121 @@
...
+- **By mouse**: press Ctrl+Shift and drag an element to clone it.
+
+- **By keyboard**: select an element and press Ctrl+Shift+arrow key to create a copy next to the original element in the arrow's direction.
+
+![Clone a Shape](../../images/diagram-clone-element.gif)
+
+
+## Link a Connector to a Shape
+
+1. Hover the mouse cursor over a shape to display its connection points.
```
Comment (dirk-pieterse) [inline interface-elements-for-web/articles/diagram/work-with-chart-elements.md:60]:
```suggestion
1. Hover the mouse pointer over a shape to display its connection points.

```

File: interface-elements-for-web/articles/diagram/work-with-chart-elements.md [hunk 17]
```diff
@@ -0,0 +1,121 @@
...
+A connector can have several texts. A text box appears at the double-clicked position. After editing, you can use the mouse to move the text along the connector.
+
+![Edit Text](../../images/diagram-edit-text.gif)
+
+## Clone an Element
+
+You can use the mouse or keyboard commands to create a copy of an element or a group of [selected elements](select-elements.md).
+
+
+- **By mouse**: press Ctrl+Shift and drag an element to clone it.
```
Comment (dirk-pieterse) [inline interface-elements-for-web/articles/diagram/work-with-chart-elements.md:51]:
```suggestion
- **Mouse**: press Ctrl+Shift and drag an element to clone it.

```

File: interface-elements-for-web/articles/diagram/work-with-chart-elements.md [hunk 18]
```diff
@@ -0,0 +1,121 @@
...
+
+- **By keyboard**: select an element and press Ctrl+Shift+arrow key to create a copy next to the original element in the arrow's direction.
```
Comment (dirk-pieterse) [inline interface-elements-for-web/articles/diagram/work-with-chart-elements.md:53]:
```suggestion
- **Keyboard**: select an element and press Ctrl+Shift+arrow to create a copy next to the original element in the arrow's direction.

```

File: interface-elements-for-web/articles/diagram.md [hunk 19]
```diff
@@ -0,0 +1,19 @@
+---
+title: Diagram
+author: Svetlana Nikulina
+---
+# Diagram
+
+The **Diagram** allows you to design diagrams, flowcharts, and organizational charts online.
```
[the remark this answers is not in the document]
Comment (LanaDX) [reply interface-elements-for-web/articles/diagram.md:7]: added "tool"

File: interface-elements-for-web/articles/diagram/export-to-an-image.md [hunk 20]
```diff
@@ -0,0 +1,14 @@
...
+author: Svetlana Nikulina
+---
+# Export to an Image
+
+The **Diagram** can export its content to the following image formats: 
+- PNG
+- JPEG
+- SVG
+
+Click the **Export** command on the **View** toolbar and select a format to download the document content in this image format.
```
[the remark this answers is not in the document]
Comment (LanaDX) [reply interface-elements-for-web/articles/diagram/export-to-an-image.md:12]: Updated to:
```suggestion
Click the **Export** command on the **View** toolbar and select a format to download the document.

```

File: interface-elements-for-web/articles/diagram/first-look-at-the-diagram.md [hunk 21]
```diff
@@ -0,0 +1,56 @@
...
+title: First Look at the Diagram
+author: Svetlana Nikulina
+---
+# First Look at the Diagram
+
+![Tools](../../images/diagram-tools.png)
+
+## Diagram Tools
+
+- The **History** toolbar allows you to undo and redo actions. On mobile devises, the toolbar also displays the **Show Toolbox** button that shows/hides the **Toolbox**.
```
[the remark this answers is not in the document]
Comment (LanaDX) [reply interface-elements-for-web/articles/diagram/first-look-at-the-diagram.md:11]:
```suggestion
- The **History** toolbar contains **Undo** and **Redo** commands that allow you to undo and redo actions. On mobile devises, the toolbar also displays the **Show Toolbox** button that shows/hides the **Toolbox**.

```

File: interface-elements-for-web/articles/diagram/first-look-at-the-diagram.md [hunk 22]
```diff
@@ -0,0 +1,56 @@
...
+
+    ![Show Toolbox Button](../../images/diagram-history-and-toolbox-button.png)
+
+- The **Toolbox** allows you to add shapes to your chart. Select the **View** > **Settings** > **Show Toolbox** command to switch the **Toolbox** visibility. On mobile devises, the **Show Toolbox** button is additionally displayed in the **History** toolbar.
+
+- The **Properties** toolbar allows you to [customize shape and connector style settings](create-a-new-chart.md#customize-shape-and-connector-style-settings), change the [chart layout](chart-layout.md), and specify [page settings](page-and-view-settings.md). Click the **Properties** button to show and hide the toolbar.
```
[the remark this answers is not in the document]
Comment (LanaDX) [reply interface-elements-for-web/articles/diagram/first-look-at-the-diagram.md:18]: Updated:
The **Properties** toolbar allows you to customize shape and connector style settings, change the chart layout, and specify page settings. Click the **Properties** button to display the toolbar.

File: interface-elements-for-web/articles/diagram/chart-layout.md [hunk 23]
```diff
@@ -0,0 +1,19 @@
+---
+title: Chart Layout
+author: Svetlana Nikulina
+---
+# Chart Layout
+The **Diagram** can create a chart layout based on a predefined algorithm.
+
+- The **Tree** layout algorithm is used to draw hierarchical diagrams. It arranges shapes in a tree structure and aligns them based on their hierarchical level.
+
+- The **Layered** layout algorithm is used to draw layered graphs. It arranges shapes in horizontal rows or layers and minimizes the number of connector crossings.
```
[the remark this answers is not in the document]
Comment (LanaDX) [reply interface-elements-for-web/articles/diagram/chart-layout.md:10]: Updated:
The **Layered** layout algorithm is used to draw layered graphs. It arranges shapes in horizontal rows or layers to reduce the number of connector crossings.

File: interface-elements-for-web/articles/diagram/work-with-chart-elements.md (hunk 11 above)
Comment (LanaDX) [reply interface-elements-for-web/articles/diagram/work-with-chart-elements.md:125]: Updated:
To remove a key point, move it between neighboring points.

File: interface-elements-for-web/articles/diagram/first-look-at-the-diagram.md (hunk 10 above)
Comment (LanaDX) [reply interface-elements-for-web/articles/diagram/first-look-at-the-diagram.md]: Updated:
You can use the Context Toolbox to create a shape at the end of a connector.

File: interface-elements-for-web/articles/diagram/chart-layout.md [hunk 24]
```diff
@@ -0,0 +1,19 @@
+---
+title: Chart Layout
+author: Svetlana Nikulina
+---
+# Chart Layout
+The **Diagram** tool creates a chart layout based on a predefined algorithm.
```
Comment (dirk-pieterse) [inline interface-elements-for-web/articles/diagram/chart-layout.md:6]:
```suggestion
The **Diagram** tool creates a chart layout based on predefined algorithms:

```

**Pull request merged**